### PR TITLE
fix: Bad feature values are now shown as NaN instead of Infinity

### LINKED ISSUE #329: `NaN` in feature data is shown as `Infinity` once loaded. (also, UI will probably crash if `Infinity` is given as a data value.)
## Description

`NaN` values in feature data are replaced with `Infinity` on the UI.

image

image

On closer inspection, it looks like it's caused by these lines in `JsonArrayLoader.ts`:
```
const nanToNull = (json: string): string => json.replace(/NaN/g, "null");

export default class JsonArrayLoader implements IArrayLoader {
  async load(url: string): Promise<JsonArraySource> {
    const response = await fetch(url);
    const text = await response.text();
    const { data, min, max }: FeatureDataJson = JSON.parse(nanToNull(text));
    console.log(url, data);
    return new JsonArraySource(data, min, max);
  }
}
```

and this logic up above in the constructor:
```
      this.array = array.map((val) => (val === null ? Infinity : val));
```

`JSON.parse` doesn't parse `NaN` values, so we are currently converting `NaN` to `null` and then `null` to `Infinity`. Note that this WILL NOT handle positive or negative `Infinity` in our data, which will cause this to crash and the dataset load to fail.

A few possible fixes:
- EASY: Update the JSON parsing to also convert `Infinity` or other values to `null`, and change `null` to be parsed as `NaN` by default.
- HARDER: Use a more robust JSON parsing library, like https://json5.org/, so we can directly represent `Infinity`/`NaN`/etc.

I'd also like to add unit tests for both cases!

## Expected Behavior

- Feature data values should match what is shown on the UI.

## Reproduction

https://dev-aics-dtp-001.int.allencell.org/nucmorph-colorizer/dist/index.html#/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Fnucmorph-apr2024%2Fcollection.json&dataset=Mama+Bear&feature=time

## Environment

_Any additional information about your environment. E.g. OS version, python version, browser etc._

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #329, a bug where `NaN` values are shown as `Infinity`.

*Estimated review size: tiny, 5 minutes*

Solution
========
- Added a comment explaining why `Infinity` values are stored internally.
- Added a check that replaces `Infinity` values with `NaN` on the hover tooltip.
- Deleted unused shader.
- Minor formatting changes to shader code.

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
1. Open the preview link and hover over the cells. Greyed-out cells should appear as `NaN` instead of `Infinity`: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-377/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Fnucmorph-apr2024%2Fcollection.json&dataset=Mama+Bear&feature=time&t=396
2. Compare with main build: https://allen-cell-animated.github.io/timelapse-colorizer/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Fnucmorph-apr2024%2Fcollection.json&dataset=Mama+Bear&feature=time&t=396
3. 
image

### CODE REVIEW

File: src/colorizer/shaders/colorize_RGBA8U.frag [hunk 1]
```diff
@@ -1,4 +1,5 @@
 precision highp usampler2D;
+precision highp sampler2D;
```
Comment (toloudis) [inline src/colorizer/shaders/colorize_RGBA8U.frag:2]: I suggest removing this until we have a PR where it actually matters

File: src/colorizer/shaders/colorize_R32.frag [hunk 2]
```diff
@@ -60,7 +60,7 @@ void main() {
 
   // Data buffer starts at 0, non-background segmentation IDs start at 1
   float featureVal = getFeatureVal(id - 1);
-  uint outlierVal = getOutlierVal(id - 1)
+  uint outlierVal = getOutlierVal(id - 1);
```
Comment (toloudis) [inline src/colorizer/shaders/colorize_R32.frag:63]: ??? how was this shader even compiling without the semicolon

**Review by toloudis — APPROVED** [review]
The main changes look good! But I bet you can remove the modifications from at least 1 and maybe 2 of the files

File: src/colorizer/shaders/colorize_R32.frag (hunk 2 above)
Comment (toloudis) [reply src/colorizer/shaders/colorize_R32.frag:63]: If this is a totally unused shader then maybe we should just delete the whole file

Comment (ShrimpCryptid) [reply src/colorizer/shaders/colorize_R32.frag:63]: Currently unused, yeah. I'll make that change.

**Review by interim17 — APPROVED** [review]

**Pull request merged**